corehq/apps/userreports/tasks.py

```diff
@@ -138,8 +146,67 @@ def rebuild_indicators_in_place(indicator_config_id, initiated_by=None, source=N
             config.meta.build.rebuilt_asynchronously = False
             config.save()
 
-        adapter.build_table(initiated_by=initiated_by, source=source)
-        _iteratively_build_table(config, in_place=True)
+        rows_count_before_rebuild = _get_rows_count_from_existing_table(adapter) if adapter.table_exists else None
+        try:
+            adapter.build_table(initiated_by=initiated_by, source=source)
+            _iteratively_build_table(config, in_place=True)
+        except Exception:
+            _report_ucr_rebuild_metrics(config, source, 'rebuild_datasource_in_place', adapter,
+                                        rows_count_before_rebuild, error=True)
+            raise
+        _report_ucr_rebuild_metrics(config, source, 'rebuild_datasource_in_place', adapter,
+                                    rows_count_before_rebuild)
+
+
+def _get_rows_count_from_existing_table(adapter):
+    table = adapter.get_existing_table_from_db()
+    return adapter.session_helper.Session.query(table).count()
+
+
+def _report_ucr_rebuild_metrics(config, source, action, adapter, rows_count_before_rebuild, error=False):
+    if source not in ('edit_data_source_rebuild', 'edit_data_source_build_in_place'):
+        return
+    try:
+        _report_metric_number_of_days_since_first_build(config.domain, config.get_id, action)
+        if error:
+            _report_metric_rebuild_error(config, action)
+        if not error:
+            _report_metric_increase_in_rows_count(config, action, adapter, rows_count_before_rebuild)
+    except Exception:
+        pass
+
+
+def _report_metric_number_of_days_since_first_build(domain, config_id, action):
+    try:
+        earliest_entry = DataSourceActionLog.objects.filter(
+            domain=domain,
+            indicator_config_id=config_id,
+            action__in=[DataSourceActionLog.BUILD, DataSourceActionLog.REBUILD]
+        ).earliest('date_created')
+    except DataSourceActionLog.DoesNotExist:
+        pass
+    else:
+        no_of_days = (datetime.utcnow() - earliest_entry.date_created).days
+        metrics_gauge(f'commcare.ucr.{action}.days_since_first_build', no_of_days, tags={'domain': domain})
+
+
+def _report_metric_rebuild_error(config, action):
+    from corehq.apps.userreports.views import _number_of_records_to_be_iterated_for_rebuild
```

Comment:
Nit: Does this give a circular import error when it's added to the top of the file? If not, it could rather be put there.

Reply:
Yes placed it here because of circular import